pkg/operator/controllers/guardrails/policies/README.md

```diff
@@ -0,0 +1,305 @@
+## Policy structure
+Each policy contains 2 parts, [ConstraintTemplate](https://open-policy-agent.github.io/gatekeeper/website/docs/constrainttemplates/) and [Constraint](https://open-policy-agent.github.io/gatekeeper/website/docs/howto/#constraints)
+
+ConstraintTemplate, ie,. gktemplate/\$TEMPLATE_NAME.yaml, is generated by policies/scripts/generate.sh, who combines the yaml part, ie,. the gktemplates-src/\$TEMPLATE_NAME/\$TEMPLATE_NAME.tmpl and rego program (gktemplates-src/\$TEMPLATE_NAME/src.rego).
+please dont edit the ConstraintTemplate directly, just provide the corresponding $TEMPLATE_NAME.tmpl and src.rego, the generate.sh will produce the ConstraintTemplate file.
+
+Constraint is manually created
+
+## Create new policy
+
+* Create a new subfolder for each new Constraint Template under gktemplates-src
+* Create a tmpl file with unique and meaningful name in above subfolder, which contains everything except for the rego, example:
+
+```yaml
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: aroprivilegednamespace
+  annotations:
+    metadata.gatekeeper.sh/title: "Privileged Namespace"
+    metadata.gatekeeper.sh/version: 1.0.0
+    description: >-
+      Disallows creating, updating or deleting resources in privileged namespaces.
+      including, ["^kube.*|^openshift.*|^default$|^redhat.*|^com$|^io$|^in$"]
+spec:
+  crd:
+    spec:
+      names:
+        kind: AROPrivilegedNamespace
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          type: object
+          description: >-
+            Disallows creating, updating or deleting resources in privileged namespaces.
+            including, ["^kube.*|^openshift.*|^default$|^redhat.*|^com$|^io$|^in$"]
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+{{ file.Read "gktemplates-src/aro-deny-privileged-namespace/src.rego" | strings.Indent 8 | strings.TrimSuffix "\n" }}
+```
+
+
+* Create the src.rego file in the same folder, howto https://www.openpolicyagent.org/docs/latest/policy-language/, example:
+```
+package aroprivilegednamespace
+
+violation[{"msg": msg, "details": {}}] {
+  input_priv_namespace(input.review.object.metadata.namespace)
+  msg := sprintf("Operation in privileged namespace %v is not allowed", [input.review.object.metadata.namespace])
+}
+
+input_priv_namespace(ns) {
+  any([ns == "default",
+  ns == "com",
+  ns == "io",
+  ns == "in",
+  startswith(ns, "openshift"),
+  startswith(ns, "kube"),
+  startswith(ns, "redhat")])
+}
+```
+* Create src_test.rego for unit tests in the same foler, which will be called by test.sh, howto https://www.openpolicyagent.org/docs/latest/policy-testing/, example:
+```
+package aroprivilegednamespace
+
+test_input_allowed_ns {
+  input := { "review": input_ns(input_allowed_ns) }
+  results := violation with input as input
+  count(results) == 0
+}
+
+test_input_disallowed_ns1 {
+  input := { "review": input_review(input_disallowed_ns1) }
```

Comment:
I've never looked at rego before, so pardon if I'm way off base, but should this be a call to `input_ns` vs `input_review`, which I don't see defined?